.githooks/commit-msg

```diff
@@ -1,12 +1,45 @@
 #!/bin/sh
 
-./scripts/commit-filter-check.sh && rc=$? || rc=$?
-echo return code : $rc
-if [ $rc == 1 ]
-then
-	echo "Script return code 1 so commit failed"
-	exit 1
-else
-	echo "No error returned so commit successful"
-fi
+commit_message_check (){
+      # gets the content of the current commit message
+      gitmessage="$(cat $1)"
+      
+      matchtypes="feat\|fix\|docs\|breaking\|build\|ci\|perf\|refactor\|style\|test"
+      types="feat, fix, docs, breaking, build, ci, perf, refactor, style, test"
+      
+      messagecheck=`echo $gitmessage | grep -w "$matchtypes"`
 
+      if [ -z "$messagecheck" ]
```

Comment:
@craicoverflow What is the reasoning for removing `commit-filter-check.sh` and adding the logic in here?

Reply:
agree with David here commit-filter-check.sh is still in use in circle ci 

![image](https://user-images.githubusercontent.com/16667688/55216069-b2ca0e80-51f3-11e9-8c4f-41eed6796099.png)

but it logic has not been updated to include the new filters

Reply:
@davidffrench @austincunningham I've explained this in the description:

> Also I noticed that the commit-msg hook was checking previous commits but not the current commit.
Because of this, the current commit message was passing and getting committed even if the format was incorrect. During the next commit the hook was failing on that previous commit. I changed the script to only look at the content of the current commit message.

This is why I help the two separate. The existing script will validate all commits, but the Git hook is only used to validate the current commit message.

Reply:
This makes sense, I am concerned about the logic existing in two places. Could you explain why the `commit-filter-check.sh` is working correctly for the latest commit through the CI but not from the git hook?

Reply:
I know and I considered doing a refactor but didn't in the interest of time. I intended for this to be a quick change :grimacing: I can still look at refactoring it now but not sure if it is a priority.

Reply:
`commit-filter-check` looks at committed commits. The Git hook checks a commit **message** before it is committed.

Reply:
agree this is not a priority